services/hanke-service/src/main/kotlin/fi/hel/haitaton/hanke/liitteet/AttachmentServiceImpl.kt

```diff
@@ -0,0 +1,96 @@
+package fi.hel.haitaton.hanke.liitteet
+
+import fi.hel.haitaton.hanke.HankeRepository
+import fi.hel.haitaton.hanke.currentUserId
+import fi.hel.haitaton.hanke.getCurrentTimeUTCAsLocalTime
+import java.util.*
+import javax.persistence.EntityNotFoundException
+import org.springframework.orm.jpa.JpaObjectRetrievalFailureException
+import org.springframework.transaction.annotation.Transactional
+import org.springframework.web.multipart.MultipartFile
+
+private val supportedFiletypes =
+    setOf(
+        "application/pdf",
+        "application/msword",
+        "application/vnd.openxmlformats-officedocument.wordprocessingml.document",
+        "image/jpeg",
+        "image/png",
+        "image/vnd.dwg"
+    )
+
+private const val FILESIZE_MAXIMUM_MEGABYTES = 10
+
+class AttachmentServiceImpl(
+    val hankeRepository: HankeRepository,
+    val attachmentRepository: AttachmentRepository,
+) : AttachmentService {
+    @Transactional
+    override fun add(hankeTunnus: String, liite: MultipartFile): HankeAttachment {
+        if (!supportedFiletypes.contains(liite.contentType)) {
```

Comment:
_The Content-Type for uploaded files is provided by the user, and as such cannot be trusted, as it is trivial to spoof. Although it should not be relied upon for security, it provides a quick check to prevent users from unintentionally uploading files with the incorrect type._

https://cheatsheetseries.owasp.org/cheatsheets/File_Upload_Cheat_Sheet.html

Reply:
https://www.baeldung.com/java-file-mime-type

But then again, I don't know if this is over engineering things. If we can trust the incoming payload, I guess we can just use the contentType.

Reply:
The payload content checking should be done separately. I added issue HAI-1612 to handle it.